Source/ARTJsonLikeEncoder.m

```diff
@@ -427,6 +427,7 @@ - (ARTTokenRequest *)tokenRequestFromDictionary:(NSDictionary *)input error:(NSE
     [_logger verbose:@"RS:%p ARTJsonLikeEncoder<%@>: tokenRequestFromDictionary %@", _rest, [_delegate formatAsString], input];
 
     if (![input isKindOfClass:[NSDictionary class]]) {
+        *error = [ARTErrorInfo createWithCode:ARTStateRequestTokenFailed message:@"input provided has an invalid TokenRequest"];
```

Comment:
If the method that uses `authUrl` ([this one](https://github.com/ably/ably-ios/blob/3e1832a76522348297b7564dd324103a36c6b782/Source/ARTAuth.m#L216)) expects the return to not be nil, then it should be that method the one that checks it, not the decoder. Arguably a nil input being decoded to a nil TokenRequest is correct behavior.